### DISCUSSION TITLE
How to use the new sidplayfp.filter() method?

### DISCUSSION THREAD

**Discussion by radioactiveman** [libsidplayfp/libsidplayfp#166] [Q&A] [answered]
libsidplayfp 2.10.0 has deprecated the `filter()` method from `sidbuilder.h`.

I would like to update the [SID plugin](https://github.com/audacious-media-player/audacious-plugins/tree/master/src/sid) of [Audacious](https://audacious-media-player.org/) to use the new method. The relevant line is found [here](https://github.com/audacious-media-player/audacious-plugins/blob/a5bf4556e2327a8943a57d554ccdbb514a62c6c1/src/sid/xs_sidplay2.cc#L102).
However I do not understand how to specify the `sidNum` parameter.

- Where is this integer coming from?
- Is it required to call `filter()` in a for loop now?
- Is it correct to move`filter()` after the `config()` call in line 137?

Thanks for your help and code suggestions.

```diff
diff --git a/src/sid/xs_sidplay2.cc b/src/sid/xs_sidplay2.cc
index 530aed796..5482414c2 100644
--- a/src/sid/xs_sidplay2.cc
+++ b/src/sid/xs_sidplay2.cc
@@ -99,7 +99,7 @@ bool xs_sidplayfp_init()
         return false;
     }
 
-    state.currBuilder->filter(xs_cfg.emulateFilters);
+    state.currEng->filter(0, xs_cfg.emulateFilters); // ???
     if (!state.currBuilder->getStatus()) {
         AUDERR("reSID->filter(%d) failed.\n", xs_cfg.emulateFilters);
         return false;
```

**Comment by drfiemost** [comment, marked as answer]
> * Where is this integer coming from?

It's the chip number, now you can control the filter for each chip separately, 0 for the first SID and 1/2 for the second and third if enabled

> * Is it required to call `filter()` in a for loop now?

No, only when you want to toggle the filter on or off 

> * Is it correct to move`filter()` after the `config()` call in line 137?

Yes, otherwise it has no effect (see [2d71b0d6d03d9beedccf484a479e6d04835d0ece](https://github.com/libsidplayfp/sidplayfp/commit/2d71b0d6d03d9beedccf484a479e6d04835d0ece))

**Comment by radioactiveman** [comment]
Thanks, this was helpful. Also found https://github.com/libsidplayfp/sidplayfp/commit/abcadfefc90accf32caad5fcba31c346327906ec now with a version check.

I didn't know the chip number can only be 0,1,2 and thought too complicated. Like using a for loop similar to:

```c
for (int i = 0; i < sidChips() /* or something else */; i++)
    filter(i, enabled);
```
